File: Javascript/Algorithms/119.js
let getRow = n => {
    const result = [1],
          floor = Math.floor,
          round = Math.round;
    let fl = floor((n+1) / 2);
    let [mid, odd] = (n + 1) & 1 ? [fl + 1, true] : [fl, false];
        
    for (let k = 1; k < mid; k++) {
        let prev = result[k-1];
        result.push(round(prev * ((n + 1 - k) / k)));
    }
    // Note: reverse is in-place, need to slice *first*.
    return result.concat(odd? result.slice(0, -1).reverse() : result.slice().reverse());    
};

File: C/Algorithms/119.c
#include <math.h>
#include <stdlib.h>

int *getRow(int n, int *returnSize) {
  int *result = malloc((n + 1) * sizeof(*result));
  *returnSize = n + 1;
  result[0] = 1;

  // Use symmetry of rows.
  int odd = 0, mid;
  if ((n + 1) & 1) {
    odd = 1;
    mid = ((n + 1) / 2 + 1);
  } else {
    mid = ((n + 1) / 2);
  }
  // Fill until mid point.
  for (int k = 1; k < mid; k++) {
    int previous = result[k - 1];
    // Need double cast, else precision costs.
    result[k] = round(previous * ((double)(n + 1 - k) / k));
  }
  // Manually reverse and add to end of result.
  int end = odd ? mid - 1 : mid;
  for (int i = 0; i < end; i++) {
    result[n--] = result[i];
  }
  return result;
}
